src/core/services/layout/layout.js

```diff
@@ -1,319 +1,414 @@
-(function () {
+(function() {
   'use strict';
 
-  var $mdUtil, $$mdLayout, $parse, $interpolate;
+  var $mdUtil, $$mdLayout, $interpolate;
 
+  var SUFFICES = /(-sm|-gt-sm|-md|-gt-md|-lg|-gt-lg)/g;
+  var WHITESPACE = /\s+/g;
+
+  var FLEX_OPTIONS = ['grow', 'initial', 'auto', 'none'];
+  var LAYOUT_OPTIONS = ['row', 'column'];
+  var ALIGNMENT_OPTIONS = [
+        "start start", "start center", "start end",
+        "center", "center center", "center start", "center end",
+        "end", "end center", "end start", "end end",
+        "space-around", "space-around center", "space-around start", "space-around end",
+        "space-between", "space-between center", "space-between start", "space-between end"
+      ];
```

Comment:
Instead of formatting feedback, I was hoping more for logic and refactoring feedback and manual testing.... ;-)